src/Logger.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Component;
+
+use Monolog\Formatter\LineFormatter;
+use Monolog\Handler\GelfHandler;
+use Monolog\Handler\HandlerInterface;
+use Monolog\Handler\StreamHandler;
+use Monolog\Logger as MonologLogger;
+use function array_filter;
+use function array_values;
+
+class Logger extends MonologLogger
+{
+    /**
+     * @param HandlerInterface[] $handlers
+     */
+    public function __construct(
+        array $handlers = []
+    ) {
+        parent::__construct('php-component');
+
+        // only keep valid handlers
+        $handlers = array_filter(
+            array_values($handlers),
+            function ($handler) {
+                return $handler instanceof HandlerInterface;
+            }
+        );
+
+        // no handlers
+        if (count($handlers) === 0) {
+            $errorHandler = self::getDefaultErrorHandler();
+            $logHandler = self::getDefaultLogHandler();
+
+            $handlers = [
+                $errorHandler,
+                $logHandler,
+            ];
+        }
+
+        // gelf log handler
+        if (count($handlers) === 1 && !($handlers[0] instanceof GelfHandler)) {
+            throw new \Exception(
+                'If only one handler is provided, it needs to be GelfHandler'
+            );
+        }
+
+        $this->setHandlers($handlers);
+    }
+
+    public static function getDefaultErrorHandler(): StreamHandler
+    {
+        $errorHandler = new StreamHandler('php://stderr');
+        $errorHandler->setBubble(false);
+        $errorHandler->setLevel(MonologLogger::WARNING);
+        $errorHandler->setFormatter(new LineFormatter("%message%\n"));
```

Comment:
Na tomhle jsem se teď pěkně zasekal https://github.com/keboola/db-extractor-mssql/pull/29/files#r188010964
Měl by tu imho být default formatter který nastavuje formát `[%datetime%] %channel%.%level_name%: %message% %context% %extra%`. nebo nějaký upravený ale je důležitý ten kontext.

Platí to teda hlavně pokud se loguje takhle https://github.com/keboola/db-extractor-mssql/blob/2372e5277685bad810391d6d292b6f61d3da1570/src/run.php#L60 což je tuším u většine extractorů.

Pokud se loguje takhle https://github.com/keboola/component-generator/blob/a9baa26b993c8a45190c7c27812da68e07378c0d/templates/php-component/src/run.php#L17 je to ok.

Reply:
založil jsem issue k diskuzi https://github.com/keboola/php-component/issues/29

Reply:
Jasně - ten "errorLog" padá celý do papertrailu pak. Takže tam dává smysl mít k tomu i ty další věci. Ale pro ten "log" je ta jen message správně, ne?